compiler/src/Language/Bond/Codegen/Cpp/Types_h.hs

```diff
@@ -86,11 +87,16 @@ types_h userHeaders enumHeader allocator alloc_ctors_enabled cpp file imports de
 
     anyNullable = Any . isNullable
 
+    anyStringOrContainer t
+        | isString t || isMetaName t || isContainer t = Any True
+        | otherwise = Any False
+
     bondHeaders :: [(Bool, String)]
     bondHeaders = [
         (have anyNullable, "<bond/core/nullable.h>"),
         (have anyBonded, "<bond/core/bonded.h>"),
-        (have anyBlob, "<bond/core/blob.h>")]
+        (have anyBlob, "<bond/core/blob.h>"),
+        (scoped_alloc_enabled && have anyStringOrContainer, "<scoped_allocator>")]
```

Comment:
What about nullable/maybe fields? Should they use the scoped allocator or not?

Reply:
The `bond::maybe` is forwarding the allocator to its element and does not directly use it, so no changes there. The `bond::nullable` however uses it in some cases, but since `std::scoped_allocator_adaptor` is compatible with its outer one, no changes there too.